modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -704,6 +704,13 @@ private Object _getValue(
 				_log.debug(portalException);
 			}
 
+			if (Objects.equals(entityType, EntityField.Type.ID) &&
+				Objects.equals(DBManagerUtil.getDBType(), DBType.POSTGRESQL) &&
+				Validator.isNumber(String.valueOf(right))) {
+
+				return GetterUtil.getLong(right);
+			}
```

Comment:
@icgsbr I won't have time to review the test code for the next three days, so please **don't forward** this PR yet. However, it is okay to create a hotfix with this change. Thanks!